api/src/integrations/nris-emli/datasource.js

```diff
@@ -0,0 +1,466 @@
+          } else {
+            if (this.shouldRecordHaveAttachments(newRecord)) {
+              await this.createRecordAttachments(records[i], newRecord);
+            }
+            await this.createItem(newRecord);
+          }
+
+          // Assuming we didn't get thrown an error, update the items successfully processed.
+          processingObject.itemsProcessed++;
+        } else {
+          // We skip and don't include non-completed records
+          processingObject.itemTotal--;
+        }
+      }
+      processingObject.status = 'Completed';
+      processingObject.message = 'updateRecordType - all done.';
+    } catch (error) {
+      processingObject.status = 'Failed';
+      processingObject.message = 'updateRecords - unexpected error: ' + error;
+      defaultLog.error(`updateRecords - unexpected error: ${error.message}`);
+    }
+    defaultLog.info('returning', processingObject.status);
+    return processingObject;
+  }
+
+  async transformRecord(record) {
+    const Inspection = mongoose.model(RECORD_TYPE.Inspection._schemaName);
+    let newRecord = new Inspection().toObject();
+    // We don't need this as we insert based on assessmentId
+    delete newRecord._id;
+
+    const legislation = {
+      act: 'Mines Act',
+      section: 15
+    };
+
+    newRecord.legislationDescription = 'Inspection to verify compliance with regulatory requirements.';
+    newRecord.recordType = 'Inspection';
+    newRecord._sourceRefNrisId = record.assessmentId;
+    try {
+      newRecord.dateIssued = record.inspection.inspctReportSentDate
+        ? moment.tz(record.inspection.inspctReportSentDate, 'America/Vancouver').toDate()
+        : null;
+    } catch (error) {
+      defaultLog.debug(error);
+      newRecord.dateIssued = null;
+    }
+    newRecord.issuingAgency = 'EMLI';
+    newRecord.author = 'EMLI';
+    newRecord.legislation = { ...legislation };
+    newRecord.dateAdded = new Date();
+    newRecord.dateUpdated = new Date();
+
+    newRecord.addedBy = (this.auth_payload && this.auth_payload.displayName) || '';
+    newRecord.updatedBy = (this.auth_payload && this.auth_payload.displayName) || '';
+
+    newRecord.sourceSystemRef = 'nris-emli';
+
+    // check if document exists already in the system but is orphaned
+    for (let i = 0; i < record.attachment.length; i++) {
+      if (record.attachment[i].fileType === 'Final Report') {
+        const existingDocument = await this.findExistingDocument(record.attachment[i].filePath);
+        if (existingDocument && existingDocument._id && !newRecord.documents.includes(existingDocument._id)) {
+          newRecord.documents.push(existingDocument._id);
+        }
+      }
+    }
+
+    if (record.client && record.client.length > 0 && record.client[0]) {
+      newRecord.issuedTo = {
+        type: 'Company',
+        companyName: record.client[0].orgName || '',
+        fullName: record.client[0].orgName || ''
+      };
+    }
+
+    newRecord.recordName = record.assessmentId;
```

Comment:
Record name should be a bit more descriptive similar to FLNRO Inspections, was overlooked in the transformation planning table.
Eg. `EMLI Inspection - Record xxxyyy`

Reply:
Fixed in 827fb0d